Update Flexer.jflex

RGT91 opened the pull request:
Añadí stack y metodo para verificar INDENT y DEINDENT.

RGT91 committed:
Update Flexer.jflex

Añadí stack y metodo para verificar INDENT y DEINDENT.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Corrección método checkStack.

RGT91 committed:
Update Flexer.jflex

Eliminé un caso que no ocurría en el loop de checkStack.

RGT91 committed:
Update Flexer.jflex

Cambié el método de lugar.

RGT91 committed:
Update Flexer.jflex

Agregué el parser a los elementos de la pila.

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,46 @@
   protected int levels_aux;
   public String tokens;
+  public int cont;
+  public Stack st = new Stack();
```
creo que aquí solo deberías de declararla y en la sección de `%init{` inicializar la pila

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -56,14 +81,15 @@ IDENTIFICADOR   = [a-zA-Z_]([a-zA-Z0-9_])*
-  {RESERVADA}      { tokens += "RESERVADA("+yytext() + ") ";yybegin(YYINITIAL); }
-  {IDENTIFICADOR}      { tokens += "IDENTIFICADOR("+yytext() + ") "; yybegin(YYINITIAL);}
+  [ ]         {  cont++;}
+  [\t\f]         {  cont+=11;}
```
porque 11?

RGT91 committed:
Update Flexer.jflex

Acomodé el init de la pila.

RGT91 committed:
Update config.yml

Agregué la prueba de jflex con fizzbuzz.

RGT91 committed:
Update config.yml

Corregí un directorio que estaba mal.

RGT91 committed:
Update Flexer.jflex

Imprime los tokens.

RGT91 committed:
Update Flexer.jflex

Imprime todos los tokens.

RGT91 committed:
Update Flexer.jflex

Salta de linea entre cada token.

RGT91 committed:
Update Flexer.jflex

Modifiqué String porque imprimía doble.

RGT91 committed:
Update Flexer.jflex

Modifiqué para ver cuál es el problema del método.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Añadí un caso al método.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Si encuentra un error de identación se detiene.

RGT91 committed:
Update Flexer.jflex

Modifiqué formato de salida.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Eliminé los print para que todo se almacene en la cadena tokens.

RGT91 committed:
Update Flexer.jflex

Agregué para que imprima los DEINDENT finales.

RGT91 committed:
Update Flexer.jflex

Agregué para que imprima los DEINDENT finales.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Actualicé el número de espacios equivalentes a un tab.

RGT91 committed:
Update Flexer.jflex

Agregué la expresión regular del End of file y los DEINDENT al final del archivo.

RGT91 committed:
Update Flexer.jflex

Eliminé la exp reg de End of file.

RGT91 committed:
Update Flexer.jflex

RGT91 committed:
Update Flexer.jflex

Eliminé la cadena que escribía checkstats en tokens porque ya se escribe la identación adentro del método.

sanjorgek commented on .circleci/config.yml:
```diff
@@ -43,6 +43,9 @@ jobs:
       - run:
           working_directory: Proyectos/Proyecto_1
           command: mvn exec:java -Dexec.mainClass="lexico.Test"
+      - run:
```
esto no es necesario ya que no tenemos ningún scanner dentro
ni esperamos argumentos

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
 %line
 %unicode
+%{
+public String checkStack(Stack st, int cont){
```
el Stack ya lo tienes declarado y en init ya lo tienes inicializado
no se porque lo pones como atributo de la función

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
 %unicode
+%{
+public String checkStack(Stack st, int cont){
+  if (cont>(int)st.peek()){
```
como tu Stack no tiene tipo del contenido, ej:
`Stack<Integer>`
por eso es necesario hacer el casting de tipo

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
+    st.push(cont);
+    tokens += "INDENT("+cont+")";
+  }else if (cont<(int)st.peek()){
+    while(cont<(int)st.peek()&&(int)st.peek()!=0){
```
creo que te tienes que detener cuando sea igual porque te quedas en esa indentación

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
+    while(cont<(int)st.peek()&&(int)st.peek()!=0){
+      st.pop();
+    }
+    if((int)st.peek()==0&&cont!=0){
```
creo que no todas las veces llegas a este error
que pasa si ` st.pop() > cont > st.pop().pop()`
esto debe de marcar un error ya que la deindentación no esta en la pila

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
   protected int levels_aux;
   public String tokens;
+  public int cont;
+  public Stack st;
```
puedes declararla como `Stack<Integer> st;` para evitar el casting

sanjorgek commented on Proyectos/Proyecto_1/src/main/jflex/Flexer.jflex:
```diff
@@ -1,21 +1,48 @@
+  cont=0;
+  st = new Stack();
+  st.push(0);
 %init}
```
esta parte en la documentación dice que es como el constructor de esta clase 
así que en este momento ya construiste la pila que es un atributo de la clase
esta es la razón por la cual no necesitas pasarla como atributo a la función
